src/parameters.jl

```diff
@@ -14,13 +14,59 @@ An abstract type that represents a set of multiple parameters.
 
 abstract type AbstractParameterSet end
 
+"""Returns the number of parameters in a parameter set."""
+function length(::T) where {T <: AbstractParameterSet}
+  return length(fieldnames(T))
+end
+
+"""Set names of parameters to their fieldnames."""
+function set_names!(parameter_set::T) where {T <: AbstractParameterSet}
+  for param_name in fieldnames(T)
+    p = getfield(parameter_set, param_name)
+    p.name = string(param_name)
+  end
+end
+
+"""Returns current values of each parameter in a parameter set."""
+function values(parameter_set::T) where {T <: AbstractParameterSet}
+  values = []
+  for param_name in fieldnames(T)
+    p = getfield(parameter_set, param_name)
+    push!(values, value(p))
+  end
+  return values
+end
+
+"""Returns lower bounds of each parameter in a parameter set."""
+function lower_bounds(parameter_set::T) where {T <: AbstractParameterSet}
+  lower_bounds = []
+  for param_name in fieldnames(T)
+    p = getfield(parameter_set, param_name)
+    d = domain(p)
+    push!(lower_bounds, lower(d))
+  end
+  return lower_bounds
+end
+
+"""Function that returns upper bounds of each parameter in a parameter set."""
+function upper_bounds(parameter_set::T) where {T <: AbstractParameterSet}
+  upper_bounds = []
+  for param_name in fieldnames(T)
+    p = getfield(parameter_set, param_name)
+    d = domain(p)
+    push!(upper_bounds, upper(d))
+  end
+  return upper_bounds
+end
```

Comment:
Do you think it would be possible to do this via an in-place function?

In other words,
```
function upper_bounds!(parameter_set, vals)
  # blah blah
  return vals
end

function upper_bounds(paremeter_set)
  vals = ... # allocates the vector
  return upper_bounds!(parameter_set, vals)
end
```